### PR TITLE
Loading kafka ssl creds the correct way for clowder + refactoring

### PR DESCRIPTION
**Pull request by tahmidefaz**
# Overview

This PR allows the loading of Kafka SSL credentials the preffered way for clowderized apps and also does some refactoring around it.

For Reference:
* The `BrokerConfig` for clowderapps: [here](https://github.com/RedHatInsights/app-common-python/blob/master/schema.json#L155-L176)
* Example from `insights-ingress-go`: [here](https://github.com/RedHatInsights/insights-ingress-go/blob/master/config/config.go#L136-L153)

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -217,6 +214,14 @@ def _from_dict(self, dict, name, default):
...
         return value
 
+    def _kafka_ca(self, cacert_string):
+        cacert_filename = None
+        if cacert_string:
+            with tempfile.NamedTemporaryFile(delete=False) as tf:
+                cacert_filename = tf.name
+                tf.write(cacert_string.encode("utf-8"))
+        return cacert_filename
+
```
Comment (kruai) [inline app/config.py:224]: Feels hacky to effectively copy the cert to a temporary file. Is this the method recommended by the Clowder team?

**Review by kruai — COMMENTED** [review]
I'm a big fan of the refactoring! I just had a couple of questions on this one before I approve.

File: app/config.py [hunk 2]
```diff
@@ -102,6 +109,14 @@ def __init__(self, runtime_environment):
         self.prometheus_pushgateway = os.environ.get("PROMETHEUS_PUSHGATEWAY", "localhost:9091")
         self.kubernetes_namespace = os.environ.get("NAMESPACE")
 
+        self.kafka_ssl_configs = {
+            "security_protocol": os.environ.get("KAFKA_SECURITY_PROTOCOL", "PLAINTEXT").upper(),
+            "ssl_cafile": self.kafka_ssl_cafile,
+            "sasl_mechanism": os.environ.get("KAFKA_SASL_MECHANISM", "").upper(),
```
[the remark this answers is not in the document]
Comment (tahmidefaz) [reply app/config.py]: It doesn't really get used unless the `security_protocol` is set to `sasl_ssl`, but I changed it to `PLAIN` anyways :+1:

**Review by kruai — APPROVED** [review]
LGTM!

**Review by ryandillinfelton — APPROVED** [review]
Looks good! Love how much duplication you got rid of!

**Review by thearifismail — COMMENTED** [review]
IMO, the PR should be placed on HOLD until Clowder migration to stage is done

**Comment by kruai** [comment]
> IMO, the PR should be placed on HOLD until Clowder migration to stage is done

What's left to do?

**Comment by tahmidefaz** [comment]
> IMO, the PR should be placed on HOLD until Clowder migration to stage is done

@thearifismail @kruai Feel free to disagree. Based on the conversation over slack, the `inventory-host-synchonizer` job is something we rarely use, and shouldn't stand as a blocker for the Kafka encryption work here. I am happy to wait until early next week to see if we figure out the `host-synchronizer` issue, but waiting beyond that to merge this doesn't seem necessary imo.

**Review by thearifismail — APPROVED** [review]

**Pull request merged**